client/components/themes-list/index.jsx

```diff
@@ -82,15 +82,21 @@ export const ThemesList = ( props ) => {
 		window.location.assign( `/start/${ WITH_THEME_ASSEMBLER_FLOW }?${ params }` );
 	};
 
-	const matchingWpOrgThemes = useMemo(
-		() =>
+	const matchingWpOrgThemes = useMemo( () => {
+		const themeSlugs = props.themes.map( ( theme ) => theme.id );
+		// Avoid duplicate themes. Some free themes are available in both wpcom and wporg.
+		if ( themeSlugs.includes( props.searchTerm.toLowerCase() ) ) {
+			return [];
+		}
+		return (
 			props.wpOrgThemes?.filter(
 				( wpOrgTheme ) =>
 					wpOrgTheme?.name?.toLowerCase() === props.searchTerm.toLowerCase() ||
 					wpOrgTheme?.id?.toLowerCase() === props.searchTerm.toLowerCase()
```

Comment:
I would address this by adding a new filter below checking if the id of each item of `wpOrgThemes` matches with `themeSlugs`. As `wpOrgThemes` are also composed by name matching themes there is a possibility, there is a possibility of we be hiding a theme that is not exactly the same.

```suggestion
		return (
			props.wpOrgThemes?.filter(
				( wpOrgTheme ) =>
				    !themeSlugs.includes( wpOrgTheme?.id?.toLowerCase() ) &&
					(wpOrgTheme?.name?.toLowerCase() === props.searchTerm.toLowerCase() ||
					wpOrgTheme?.id?.toLowerCase() === props.searchTerm.toLowerCase())
```

Reply:
Thanks for catching this and the suggestion! Fixed in https://github.com/Automattic/wp-calypso/pull/75169/commits/d035d6bc0770d6ebf60c716287a1964ae9884e85